source/extensions/filters/network/mongo_proxy/proxy.cc

```diff
@@ -208,21 +216,22 @@ void ProxyFilter::decodeReply(ReplyMessagePtr&& message) {
     }
 
     if (!active_query.query_info_.command().empty()) {
-      std::string stat_prefix =
-          fmt::format("{}cmd.{}", stat_prefix_, active_query.query_info_.command());
-      chargeReplyStats(active_query, stat_prefix, *message);
+      Stats::StatNameVec names{mongo_stats_->cmd_,
+                               mongo_stats_->getStatName(active_query.query_info_.command())};
+      chargeReplyStats(active_query, names, *message);
     } else {
       // Collection stats first.
-      std::string stat_prefix =
-          fmt::format("{}collection.{}.query", stat_prefix_, active_query.query_info_.collection());
-      chargeReplyStats(active_query, stat_prefix, *message);
+      Stats::StatNameVec names{mongo_stats_->collection_,
+                               mongo_stats_->getStatName(active_query.query_info_.collection()),
+                               mongo_stats_->query_};
+      chargeReplyStats(active_query, names, *message);
 
       // Callsite stats if we have it.
       if (!active_query.query_info_.callsite().empty()) {
-        std::string callsite_stat_prefix =
-            fmt::format("{}collection.{}.callsite.{}.query", stat_prefix_,
-                        active_query.query_info_.collection(), active_query.query_info_.callsite());
-        chargeReplyStats(active_query, callsite_stat_prefix, *message);
+        names[2] = mongo_stats_->callsite_; // Replaces "query".
```

Comment:
If I'm not mistaken, `2` _should_ be the last index since `chargeReplyStats` resizes to the original size (could be validated with something like `ASSERT(names.size() == 2);`) so maybe `.back()` could be appropriate here rather than a magic number.

Reply:
Note quite, but it's definitely worth a comment showing what's happening to this array.